swan/scripts/before-notebook.d/02_environment.sh

```diff
@@ -6,111 +6,118 @@
 # launched as regular user as it's this entity which is able to access 
 # eos and not the super user.
 
-# Set environment for the Jupyter process
-LOCAL_HOME=/home/$NB_USER
-export IPYTHONDIR=$LOCAL_HOME/.ipython
-export JPY_DIR=$LOCAL_HOME/.jupyter
-
-# The Jupyter environment variables below are set to point to the local user
-# home (/home/$USER), instead of the EOS path
-export JUPYTER_CONFIG_DIR=$JPY_DIR
-export JUPYTER_PATH=$LOCAL_HOME/.local/share/jupyter
-export JUPYTER_RUNTIME_DIR=$JUPYTER_PATH/runtime
-export JUPYTER_DATA_DIR=$JUPYTER_PATH
-
-# Set npm cache on the local user home, instead of the EOS path
-export NPM_CONFIG_CACHE=$LOCAL_HOME/.npm
-
-# Set jupyterlab config directory
-export JUPYTERLAB_SETTINGS_DIR=$HOME/.local/lab/user-settings/
-
-# Set other environment variables
-export KERNEL_DIR=$JUPYTER_PATH/kernels
-export SWAN_ENV_FILE=$LOCAL_HOME/.bash_profile
-export PROFILEPATH=$IPYTHONDIR/profile_default
-
-# Create missing directories
-mkdir -p $IPYTHONDIR $PROFILEPATH
-
-# Setup the LCG View on CVMFS
-_log "Setting up environment from CVMFS"
-export LCG_VIEW=$ROOT_LCG_VIEW_PATH/$ROOT_LCG_VIEW_NAME/$ROOT_LCG_VIEW_PLATFORM
-
-# symlink $LCG_VIEW/share/jupyter/nbextensions for the notebook extensions
-ln -s $LCG_VIEW/share/jupyter/nbextensions $JUPYTER_PATH
-
-# Configure kernels and terminal
-# The environment of the kernels and the terminal will combine the view and the user script (if any)
-_log "Configuring kernels and terminal"
-
-# ROOT
-ROOT_KERNEL_PATH=$LCG_VIEW/etc/notebook/kernels/root
-if [ -d $ROOT_KERNEL_PATH ];
+if [ -n "${ROOT_LCG_VIEW_PATH}" ] && [ -n "${ROOT_LCG_VIEW_NAME}" ] && [ -n "${ROOT_LCG_VIEW_PLATFORM}" ];
```

Comment:
There will be a condition for this when custom envs are there, Rodrigo has it in his PRs.

How can this variables be missing now if we always select LCG?

Reply:
> Had to do some of these to run the image on my laptop.

Reply:
Did you perhaps go through this else:

https://github.com/swan-cern/jupyterhub-extensions/blob/master/SwanSpawner/swanspawner/swanspawner.py#L124

That would explain it. Actually I have a question about that else where I mentioned you in a PR of Rodrigo, I wonder when the variables of that else need to be set (what does it mean there "via the API" and when can this happen?).

Reply:
@etejedor I ran the image directly..... `docker run ....`

That API can be called from Binder (we could also call it, but we don't). This were tests of Prasanth but were left since we were expecting to use it at some point.

Reply:
Ok then, we can put in this change, but it will be superseded by a change later by Rodrigo. So OK to put this now, and we review what Rodrigo has in his PR.

Regarding the else, let's comment on Rodrigo's PR and potentially get rid of those variables:

https://github.com/swan-cern/jupyterhub-extensions/pull/94/files#r1628115476

Reply:
I think @RodrigoSobral2000 made the same change, but either way, what matters is that we do not configure if we do not have the proper params.